### PR TITLE
Refactor ed25519::convert_ed_keypair_to_curve25519() method

### PR DESCRIPTION
**Pull request by pavel-mukhanov**

### CODE REVIEW

File: src/crypto/sign/ed25519.rs [hunk 1]
```diff
@@ -153,21 +153,21 @@ pub fn verify_detached(
 }
 
 /// Convert Ed25519 public key to Curve25519 public key.
-pub fn convert_ed_pk_to_curve25519(pk: PublicKey) -> [u8; SCALARMULTBYTES] {
-    let pk = clone_into_array(&pk[..]);
+pub fn convert_ed_pk_to_curve25519(PublicKey(ref pk): PublicKey) -> [u8; SCALARMULTBYTES] {
```
Comment (stanislav-tkach) [inline src/crypto/sign/ed25519.rs]: Interesting, I rarely see such pattern. :upside_down_face: 

@ozkriff @vitvakatu

**Review by stanislav-tkach — APPROVED** [review]

File: src/crypto/sign/ed25519.rs (hunk 1 above)
Comment (ozkriff) [reply src/crypto/sign/ed25519.rs]: I'm not sure that pattern matching _in argument position_ is really justified in this case - `&pk.0` looks much less surprising for most Rust programmers. Do we have any code style agreement about this?

Comment (pavel-mukhanov) [reply src/crypto/sign/ed25519.rs]: Hm, this pattern was widely used in `sodiumoxide`, but now they prefer `pk.0`.

Comment (stanislav-tkach) [reply src/crypto/sign/ed25519.rs]: I prefer `pk.0` too. :upside_down_face:

Comment (vitvakatu) [reply src/crypto/sign/ed25519.rs]: Looks interesting. I know about that pattern, but I've never seen it in real projects. I'm voting for `&pk.0`

**Review by ozkriff — APPROVED** [review]

**Review by stanislav-tkach — APPROVED** [review]

**Review by aleksuss — APPROVED** [review]

**Pull request merged**